amazonec2/Module.md

```diff
@@ -37,23 +37,85 @@ First, import the `wso2/amazonec2` module into the Ballerina project.
+
+1. Follow the steps below to get an AWS Access Key for your AWS root account:
+
+    * Go to the AWS Management Console.
+    * Hover over your company name in the right top menu and click "My Security Credentials".
+    * Scroll to the "Access Keys" section.
+    * Click on "Create New Access Key".
+    * Copy both the Access Key ID (YOUR_AMAZON_EC2_KEY) and Secret Access Key (YOUR_AMAZON_EC2_SECRET).
+
+2. Follow the steps below to get an AWS Access Key for an IAM user account:
+
+    * Sign in to the AWS Management Console and open the IAM console.
+    * In the navigation pane, choose Users.
+    * Add a check mark next to the name of the desired user, and then choose User Actions from the top.
+    * Click on Manage Access Keys.
+    * Click on Create Access Key.
+    * Click on Show User Security Credentials. Copy and paste the Access Key ID and Secret Access Key values, or click on Download Credentials to download the credentials in a CSV (file).
+
+3. Obtain the following parameters
+    * Access key ID.
+    * Secret access key.
+    * Desired Server region.
+
+### Obtaining temporary security credentials using IAM roles
+
+Temporary credentials are primarily used with IAM roles. You can request temporary credentials that have a more restricted set of permissions than your standard IAM user. A benefit of temporary credentials is that they expire automatically after a set period of time. You have control over the duration that the credentials are valid. Temporary security credentials work almost identically to the long-term access key credentials that your IAM users can use.
+An application on the instance retrieves the security credentials provided by the role from the instance metadata item iam/security-credentials/role-name. The application is granted the permissions for the actions and resources that you've defined for the role through the security credentials associated with the role. These security credentials are temporary and we rotate them automatically.
+
+1. Follow this doc <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html>, to launch an EC2 Instance.
+2. Get the SSH key (.pem file) is provided by Amazon when you launch the instance.
+3. Follow this doc to connect to your instance using an SSH client <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AccessingInstancesLinux.html#AccessingInstancesLinuxSSHClient>
+4. Retrieve Security Credentials from your Instance Metadata. The following command retrieves the security credentials for an IAM role named ec2access.
+    ```
+    curl http://169.254.169.254/latest/meta-data/iam/security-credentials/ec2access
```

Comment:
What’s this IP address? Is it the IP of the instance? If so, shall we not put the actual IP of an instance

Reply:
AWS exposes an Instance Metadata endpoint on every EC2 Instance at http://169.254.169.254. If we SSH to an EC2 Instance, we can use any HTTP client, such as curl, to get information about that Instance from its Instance Metadata endpoint. It is same for all.
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html

Reply:
If I understood correctly, each instance provides an endpoint for retrieving IAM roles attached to that particular instance (`/iam/security-credentials/<role_name>`). This IP address would be the IP address of a particular EC2 instance. It's a bit misleading if you say you can retrieve info using the following command and it contains a fixed IP address. 

Also, shall we refrain from copying stuff from AWS docs verbatim?

Reply:
No, this is not the IP address of a particular EC2 instance. 
1. I have used this command to ssh to my EC2 instance. Here, ec2-54-208-89-179.compute-1.amazonaws.com will be the public_dns_name of the created EC2 instance.
```ssh -i biruntha.pem ec2-user@ec2-54-208-89-179.compute-1.amazonaws.com```

2. After logged into the EC2 instance, if I run the following command from inside the EC2 instance:
```curl http://169.254.169.254/latest/meta-data/iam/security-credentials/```
it will give the name of the role associated with the EC2 instance.

3. Now you can reference that role name in the following command:
```curl http://169.254.169.254/latest/meta-data/iam/security-credentials/role_name```

4. This will give the temporary security credentials

Sample output
```
biruntha@biruntha:~/Desktop/AmazonEC2$ ssh -i biruntha.pem ec2-user@ec2-3-94-116-5.compute-1.amazonaws.com
Last login: Mon May 13 05:22:55 2019 from 203.94.95.4

       __|  __|_  )
       _|  (     /   Amazon Linux AMI
      ___|\___|___|

https://aws.amazon.com/amazon-linux-ami/2018.03-release-notes/
15 package(s) needed for security, out of 22 available
Run "sudo yum update" to apply all updates.
[ec2-user@ip-172-31-12-162 ~]$
[ec2-user@ip-172-31-12-162 ~]$ 
[ec2-user@ip-172-31-12-162 ~]$ curl http://169.254.169.254/latest/meta-data/iam/security-credentials/
ec2RoleBiruntha2
[ec2-user@ip-172-31-12-162 ~]$ 
[ec2-user@ip-172-31-12-162 ~]$ 
[ec2-user@ip-172-31-12-162 ~]$ 
[ec2-user@ip-172-31-12-162 ~]$ curl http://169.254.169.254/latest/meta-data/iam/security-credentials/ec2RoleBiruntha2
{
  "Code" : "Success",
  "LastUpdated" : "2019-05-14T03:51:34Z",
  "Type" : "AWS-HMAC",
  "AccessKeyId" : "ASIASCC###Q7VGE",
  "SecretAccessKey" : "iQ+948JLvGslMsnabg####dx/RWg808QWTF",
  "Token" : "AgoJb3JpZ2luX2VjENT//////////#############6VF+pAnk/ltYIyB17OQlGOBKaEEJaSJR/EAYfbNXKkBp6vOLnJ6OZNc8FMCOCnvXoAAWOhNFHZzKHapooejTiJuCHRen1uRm800KALO13+99zzmTQ/YAChlhac8eGboYwXJSVnr+bERqeI63LGFjfhFxpkt2jPxONpN6Ua1dxrKBS8UAMTgYDOYebOOiYX9/1g7dhYu2G4dE/k569LXsqkj5KVF4os2hOYcW6jf9XxZZ0Y2YMWB0adYQMi98MqwBE15XTxzkJJgJTvcta3oS/ySe7dDSb8MddNPGh###########C6OiTThHgzwqlSFx7XAKcQ5wOalspbkms9admJgYZO2DAzSQ/mI2RiCapisPdHsjiBW/ihvV5xyTOqOBfqPXPqy/TD6UfNCqiERAO96IVn473XfZDdYuQgtlYRTJhD7vFKTZYic8Kg/36S5Ow0H4GYDbDf3YMtsZBUExKHMY60wVba+M=",
  "Expiration" : "2019-05-14T10:27:31Z"
}
[ec2-user@ip-172-31-12-162 ~]$ 
```

Reply:
ah ok.. then no problem